### DISCUSSION TITLE
DSL/ASL Parsing

### DISCUSSION THREAD

**Discussion by ClemaX** [rust-osdev/acpi#116] [Q&A]
Are you planning to support source file parsing, or is it outside of the scope of this project?

**Comment by IsaacWoods** [comment]
I have thought about it, but don't have a usecase or the time myself - it would probably be most useful as the parser for a full ASL-to-AML compiler, which I do think would be useful as `iasl` can be finicky, and ACPICA is not an easy codebase to read. Do you have another use for one?

So I'd say it is in-scope, and I believe having a single collaborative Rust implementation for ACPI is the way to go forward given its complexity, but I have no personal plans to write one. However, I would be more than happy to host it in the repo, as well as provide code review, if you or anyone else would like to contribute one :)
